napari/_qt/event_loop.py

```diff
@@ -83,7 +83,7 @@ def gui_qt(*, startup_logo=False, gui_exceptions=False):
     # if the application already existed before this function was called,
     # there's no need to start it again.  By avoiding unnecessary calls to
     # ``app.exec_``, we avoid blocking.
-    if app.applicationName() == 'napari':
+    if app.applicationName() in ('napari', 'magicgui'):
```

Comment:
Not to be be nitpicky, but do you want to keep
```python
COMPATIBLE_APPLICATIONS = (`napari`, `magicgui`)
```
and then make this `in COMPATIBLE_APPLICATIONS`. Might be a little cleaner more/ future proof

Reply:
I don't mind the new string approach, I thought it was quite heavyweight above...

Reply:
BUT, there should be a comment here linking to this PR and explaining that we treat magicgui applications specially because the decorator creates them even before napari has had a chance to do so.

Reply:
This would just be variable, not two like before. It would then be very easy to associate comment with that variable when it's defined and not in the middle of event loop code, but i'll defer to @tlambert03

Reply:
> * add a keyword argument to provide "valid" application names, defaults to `['napari']` but can be set to anything.
> * add an argument `any=False` that can be set to True to ignore the name check
> * add an argument to _magicgui_ to set the application name, and the above becomes `@magicgui(application='napari')`

these would all "work" ... but I feel like we're passing the burden of understanding what's happening here to users.  I can see "wait, when do I need to use `magicgui(application=napari)`?" (answer: only if you use the decorator _before_ calling `napari.gui_qt`), "when do I need to use `any=False`" (answer: be careful not to use it in a jupyter environment or you'll get blocking!" ...
while I agree this isn't the most elegant thing, it feels like the lesser of two weevils.